app/models/project.rb

```diff
@@ -1,5 +1,12 @@
 class Project < ActiveRecord::Base
-  has_and_belongs_to_many :users
+  has_many :projects_users, class_name: "ProjectsUsers",
+                            foreign_key: "project_id",
+                            dependent: :destroy
+  has_many :users, through: :projects_users
+
+  has_many :owners, -> { where(owner: true) }, class_name: 'ProjectsUsers'
+  has_many :members, -> { where(owner: false) }, class_name: 'ProjectsUsers'
```

Comment:
Something like:
 `has_many :members, -> { where('projects_users.owner = ?', false) }, through: :projects_users, source: :user`
would be more useful to read this information, since it would return User instances directly, however this wouldn't allow rails to write a user into the owner list by doing e.g. `project.owners << User.first`.